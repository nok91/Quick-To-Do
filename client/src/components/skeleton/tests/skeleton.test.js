import React from 'react';
import { render, screen } from '@testing-library/react';
import Skeleton from '../';

describe('<Skeleton />', () => {
    it('should match snapshot', () => {
        let component = render(<Skeleton />);
        expect(component.container.firstChild).toMatchSnapshot();

        component = render(
            <Skeleton>
                <Skeleton.Head>Head Content</Skeleton.Head>
                <Skeleton.Body>Body Content</Skeleton.Body>
                <Skeleton.Footer>Footer Content</Skeleton.Footer>
            </Skeleton>
        );
        expect(component.container.firstChild).toMatchSnapshot();

        expect(screen.getByText(/Head Content/i)).toBeInTheDocument();
        expect(screen.getByText(/Body Content/i)).toBeInTheDocument();
        expect(screen.getByText(/Footer Content/i)).toBeInTheDocument();
    });

});
