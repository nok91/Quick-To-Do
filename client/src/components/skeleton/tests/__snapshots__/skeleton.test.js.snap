// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Skeleton /> should have a profile image 1`] = `
<div
  class="skeleton"
/>
`;

exports[`<Skeleton /> should match snapshot 1`] = `
<div
  class="skeleton"
/>
`;

exports[`<Skeleton /> should match snapshot 2`] = `
<div
  class="skeleton"
>
  <div
    class="head"
  >
    Head Content
  </div>
  <div
    class="body"
  >
    Body Content
  </div>
  <div
    class="footer"
  >
    Footer Content
  </div>
</div>
`;
